contracts/CarbonProjects.sol

```diff
@@ -317,4 +302,12 @@ contract CarbonProjects is
     {
         return memcmp(bytes(a), bytes(b));
     }
+
+    error CallerNotAllowed();
```

Comment:
The main issue with custom errors is that Polygonscan still does not support them (whereas Etherscan does).